driver-core/src/main/com/mongodb/internal/connection/DefaultConnectionPool.java

```diff
@@ -1310,4 +1355,148 @@ Timeout timeout() {
             return timeout;
         }
     }
+
+    /**
+     * Methods {@link #start()} and {@link #runOnceAndStop()} must be called sequentially. Each {@link #start()} must be followed by
+     * {@link #runOnceAndStop()} unless {@link BackgroundMaintenanceManager} is {@linkplain #close() closed}.
+     * <p>
+     * This class implements
+     * <a href="https://github.com/mongodb/specifications/blob/master/source/connection-monitoring-and-pooling/connection-monitoring-and-pooling.rst#background-thread">
+     * CMAP background thread</a>.
+     */
+    @NotThreadSafe
+    private final class BackgroundMaintenanceManager implements AutoCloseable {
+        @Nullable
+        private final ScheduledExecutorService maintainer;
+        @Nullable
+        private Future<?> cancellationHandle;
+        private boolean initialStart;
+
+        private BackgroundMaintenanceManager() {
+            maintainer = Executors.newSingleThreadScheduledExecutor(new DaemonThreadFactory("MaintenanceTimer"));
+            cancellationHandle = null;
+            initialStart = true;
+        }
+
+        void start() {
+            if (maintainer != null) {
```

Comment:
Yes, the `maintainer` was supposed to be set to `null` if the background thread was disabled. I do remember writing code in https://github.com/mongodb/mongo-java-driver/pull/701/commits/5d28ac32cfa6933af2c716e1e8ef615d23de8536 and in https://github.com/mongodb/mongo-java-driver/pull/701/commits/b54a7638fdd5697e253a170d70c1b9ba560d5bad, yet we discovered the the former was absent from the PR during the walkthrough, and the latter was causing some recent test failures (others are covered by a newly created https://jira.mongodb.org/browse/DRIVERS-1806, which I hope will be accepted). My best guess is that I lost these changes either during merges or in some side-branches.

I apologize for not committing and pushing the missing changes from the latter commit (https://github.com/mongodb/mongo-java-driver/pull/701/commits/b54a7638fdd5697e253a170d70c1b9ba560d5bad) earlier, because it would have spared you from scratching your head with regard to this code.